src/main/java/net/geforcemods/securitycraft/blocks/reinforced/ReinforcedButtonBlock.java

```diff
@@ -0,0 +1,110 @@
+
+import net.geforcemods.securitycraft.misc.CustomModules;
+import net.geforcemods.securitycraft.misc.OwnershipEvent;
+import net.geforcemods.securitycraft.tileentity.ReinforcedButtonTileEntity;
+import net.geforcemods.securitycraft.util.ModuleUtils;
+import net.minecraft.block.AbstractButtonBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.material.Material;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.storage.loot.LootContext;
+import net.minecraftforge.common.MinecraftForge;
+
+import java.util.List;
+
+public class ReinforcedButtonBlock extends AbstractButtonBlock implements IReinforcedBlock
+{
+    public static final Block.Properties STONE_PROPERTIES = Block.Properties.create(Material.MISCELLANEOUS).doesNotBlockMovement().hardnessAndResistance(-1.0F, 6000000.0F);
+    public static final Block.Properties WOOD_PROPERTIES = Block.Properties.create(Material.MISCELLANEOUS).doesNotBlockMovement().hardnessAndResistance(-1.0F, 6000000.0F).sound(SoundType.WOOD);
+    private final Block vanillaBlock;
+    public boolean isWoodenButton;
+
+    public ReinforcedButtonBlock(boolean isWooden, Block.Properties properties, Block vb)
+    {
+        super(isWooden, properties);
+        this.isWoodenButton = isWooden;
+        this.vanillaBlock = vb;
+    }
+
+    @Override
+    public SoundEvent getSoundEvent(boolean powered)
+    {
+        if (isWoodenButton) return powered ? SoundEvents.BLOCK_WOODEN_BUTTON_CLICK_ON : SoundEvents.BLOCK_WOODEN_BUTTON_CLICK_OFF;
+        else return powered ? SoundEvents.BLOCK_STONE_BUTTON_CLICK_ON : SoundEvents.BLOCK_STONE_BUTTON_CLICK_OFF;
+    }
+
+    @Override
+    public ActionResultType onBlockActivated(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand handIn, BlockRayTraceResult p_225533_6_) {
+        TileEntity tileEntity = world.getTileEntity(pos);
+
+        if (state.get(POWERED))
+        {
+            return ActionResultType.CONSUME;
+        }
+        else if (tileEntity instanceof ReinforcedButtonTileEntity && isAllowedToPress(world, pos, (ReinforcedButtonTileEntity) tileEntity, player))
+        {
+            this.func_226910_d_(state, world, pos);
+            this.playSound(player, world, pos, true);
+            System.out.println("Success!");
+            return ActionResultType.SUCCESS;
+        }
+        return ActionResultType.CONSUME;

```

Comment:
```suggestion
if(isAllowedToPress(world, pos, (ReinforcedButtonTileEntity)world.getTileEntity(pos), player))
	super.onBlockActivated(state, world, pos, player, hand, rayTrace);
```

Reply:
This function is missing a return statement in this case.

Reply:
Yup, that's true. Looks fine how you did it now, though.